command/fs_list.go

```diff
@@ -0,0 +1,121 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+)
+
+type FSListCommand struct {
+	Meta
+}
+
+func (f *FSListCommand) Help() string {
+	helpText := `
+Usage: nomad fs-list [alloc-id] [path]
+
+	Displays the files in the alloc-dir of the given alloc id. The path 
+	is relative to the root of the alloc dir.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (f *FSListCommand) Synopsis() string {
+	return "Displays list of files of an alloc dir"
+}
+
+func (f *FSListCommand) Run(args []string) int {
+	flags := f.Meta.FlagSet("fs-list", FlagSetClient)
+	flags.Usage = func() { f.Ui.Output(f.Help()) }
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+	args = flags.Args()
+
+	if len(args) < 1 {
+		f.Ui.Error("a valid alloc id is essential")
+		return 1
+	}
+
+	allocID := args[0]
+	path := "/"
+	if len(args) == 2 {
+		path = args[1]
```

Comment:
@dadgar I think it doesn't matter. We use `filepath.join` on the server side to append the right path before the user specified path.